botocore/errorfactory.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from botocore.exceptions import ClientError
+from botocore.utils import get_service_module_name
+
+
+class BaseClientExceptions(object):
+    ClientError = ClientError
+
+    def __init__(self, code_to_exception=None):
+        """Base class for exceptions object on a client
+
+        :type code_to_exception: dict
+        :param code_to_exception: Mapping of error codes (strings) to exception
+            class that should be raised when encountering a particular
+            error code.
+        """
+        self._code_to_exception = code_to_exception
+        if code_to_exception is None:
+            self._code_to_exception = {}
+
+    def from_code(self, error_code):
```

Comment:
In the previous pull request, this was an internal method. I actually think this is quite useful as it allows users to determine how ``exception.parsed_response['Error']['Code']`` gets mapped to an exception class, which would make it easier to convert legacy style error catching to this of inspecting the dictionary to this format. Would like to get thoughts on this.